scriptmodules/ports/gmloader.sh

```diff
@@ -36,23 +42,54 @@ function build_gmloader() {
 }
 
 function install_gmloader() {
-    md_ret_files=('build/gmloader')
+    md_ret_files=(build/gmloader
+        LICENSE.md
+        README.md
+    )
+}
+
+# param1: filename of APK (either relative or absolute path)
+function addlib_gmloader() {
+    local apk="$(realpath "$1")"
+    printMsgs "console" "Adding libc++_shared.so to $(basename "$apk") ..."
+    apk_decoded_dir="${apk%.*}"
+    apktool decode --no-debug-info --no-res --no-src --output "$apk_decoded_dir" "$apk"
+    # silently assuming an ARM v7 32bit apk is provided
+    if [[ -f "$apk_decoded_dir/lib/armeabi-v7a/libc++_shared.so" ]] ; then
+        printMsgs "console" "... Done: libc++_shared.so already present in $(basename "$apk")"
+        rm -rf "$apk_decoded_dir"
+        return
+    fi
+    cp "${md_data}/libc++_shared.so" "$apk_decoded_dir/lib/armeabi-v7a/"
+    printMsgs "console" "... re-packing ..."
+    apktool build --use-aapt2 --output "$apk" "$apk_decoded_dir"
+    chown pi: "$apk"
+    rm -rf "$apk_decoded_dir"
+    printMsgs "console" "... Done: Adding libc++_shared.so to $(basename "$apk")"
 }
 
 function configure_gmloader() {
     local apkdir="$romdir/ports/droidports"
     local maldita_url="https://locomalito.com/juegos/Maldita_Castilla_ouya.apk"
-    local spelunky_url="https://github.com/yancharkin/SpelunkyClassicHD/releases/download/1.1.7/spelunky_classic_hd-android.apk"
-    local maldita_file="$apkdir/$(basename ${maldita_url})"
-    local spelunky_file="$apkdir/$(basename ${spelunky_url})"
+    local spelunky_url="https://github.com/yancharkin/SpelunkyClassicHD/releases/download/1.1.7-optimized/spelunky_classic_hd-android-armv7.apk"
```

Comment:
I can't remember why I initially chose the "final" version over the "optimized" one but I thought there was something. This hasn't given you any issues?